dothething/dothething/ClipsView.swift

```diff
@@ -12,20 +12,19 @@ struct ClipView: View {
     let clip: Clip
 
     var body: some View {
-        // thumbnail of video
-        Image(uiImage: clip.thumbnail)
-            .resizable()
-            .frame(width: 108, height: 192)
-            .cornerRadius(8)
-            .onTapGesture {
-                print("Tapped \(clip.url)")
-                Thinger.playVideo(videoUrl: clip.url)
+        GeometryReader { geometry in
+            if clip.thumbnail == UIImage() {
+                Image("Placeholder")
+                    .resizable()
+            } else {
+                Image(uiImage: clip.thumbnail)
+                    .resizable()
+                    .onTapGesture {
+                        print("Tapped \(clip.url)")
+                        Thinger.playVideo(videoUrl: clip.url)
+                    }
```

Comment:
The changes in this section replace the existing thumbnail display with a conditional check. If the `clip.thumbnail` is an empty `UIImage`, it will display the "Placeholder" image. Otherwise, it will display the actual thumbnail.

This is a good improvement, as it ensures that the UI is not left with a blank space if the thumbnail image is not available yet. The use of the `GeometryReader` also allows the thumbnail to be resized and positioned correctly within the view.

The tap gesture handling for playing the video is also maintained in the new implementation.